hack/generator/pkg/codegen/pipeline_determine_resource_ownership.go

```diff
@@ -0,0 +1,290 @@
+			}
+
+			childResourcePropertyTypeDef, err := extractChildResourcePropertyTypeDef(
+				definitions,
+				def.Name(),
+				specDef.Name(),
+				specType)
+			if err != nil {
+				return nil, err
+			}
+			if childResourcePropertyTypeDef == nil {
+				continue // This just means skip
+			}
+
+			childResourceTypeNames, err := extractChildResourceTypeNames(*childResourcePropertyTypeDef)
+			if err != nil {
+				return nil, err
+			}
+
+			err = createdUpdatedChildResourceDefinitionsWithOwner(definitions, childResourceTypeNames, def.Name(), updatedDefs)
+			if err != nil {
+				return nil, err
+			}
+
+			// Remove the resources property from the owning resource spec
+			specDef = specDef.WithType(specType.WithoutProperty(resourcesPropertyName))
+
+			updatedDefs[specDef.Name()] = specDef
+		}
+	}
+
+	setResourceGroupOwnerForResourcesWithNoOwner(definitions, updatedDefs)
+	return combineNewAndExistingDefs(definitions, updatedDefs), nil
+}
+
+func resourceSpecTypeAsObject(resourceSpecDef astmodel.TypeDefinition) (*astmodel.ObjectType, error) {
+	// There's an expectation here that the spec is a typename pointing to an object. Even if the resource
+	// uses AnyOf/OneOf to model some sort of inheritance at this point that will be rendered
+	// as an object (with properties, etc)
+	specType, ok := resourceSpecDef.Type().(*astmodel.ObjectType)
+	if !ok {
+		return nil, errors.Errorf(
+			"spec (%s) type is %T, not *astmodel.ObjectType",
+			resourceSpecDef.Name(),
+			resourceSpecDef.Type())
+	}
+
+	return specType, nil
+}
+
+func extractChildResourcePropertyTypeDef(
+	definitions astmodel.Types,
+	resourceName astmodel.TypeName,
+	resourceSpecName astmodel.TypeName,
+	specType *astmodel.ObjectType) (*astmodel.TypeDefinition, error) {
+
+	// We're looking for a magical "Resources" property - if we don't find
+	// one just move on
+	resourcesProp := findResourcesProperty(specType)
+	if resourcesProp == nil {
+		return nil, nil
+	}
+
+	// The resources property should be an array
+	resourcesPropArray, ok := resourcesProp.PropertyType().(*astmodel.ArrayType)
+	if !ok {
+		return nil, errors.Errorf(
+			"Resource %s has spec %s with Resources property whose type is %T not array",
+			resourceName,
+			resourceSpecName,
+			resourcesProp.PropertyType())
+	}
+
+	// We're really interested in the type of this array
+	resourcesPropertyTypeName, ok := resourcesPropArray.Element().(astmodel.TypeName)
+	if !ok {
+		return nil, errors.Errorf(
+			"Resource %s has spec %s with Resources property whose type is array but whose inner type is not TypeName, instead it is %T",
+			resourceName,
+			resourceSpecName,
+			resourcesPropArray.Element())
+	}
+
+	resourcesDef, ok := definitions[resourcesPropertyTypeName]
+	if !ok {
+		return nil, errors.Errorf("couldn't find definition Resources property type %s", resourcesPropertyTypeName)
+	}
+
+	return &resourcesDef, nil
+}
+
+func handleObjectTypeResourcesProperty(
```

Comment:
`handle` is vague - how about `resolveResourcesTypeProperties`? Although that's still fairly vague.

Reply:
Changed this to `resolveResourcesTypeNames`